packages/medusa/src/api/middlewares/batch-job/get-requested-batch-job.ts

```diff
@@ -0,0 +1,16 @@
+import BatchJobService from "../../../services/batch-job"
+
+export async function getRequestedBatchJob(req, res, next) {
+  const id = req.params.id
+  const batchJobService: BatchJobService = req.scope.resolve("batchJobService")
+
+  let error
+  req.batch_job = await batchJobService
+    .retrieve(id)
+    .catch((err) => (error = err))
+
+  if (error) {
+    return next(error)
+  }
+  next()
+}
```

Comment:
suggestion: Use @pKorsholm change to other PR:
```suggestion
export async function getRequestedBatchJob(req, res, next) {
  const id = req.params.id
  const batchJobService: BatchJobService = req.scope.resolve("batchJobService")

  try {
    req.batch_job = await batchJobService.retrieve(id)
  } catch (error) {
    return next(error)
  }

  next()
}
```